### ISSUE TITLE
build error while installing cutorch

### ISSUE THREAD

**Issue by Maurdekye** [jcjohnson/neural-style#141]
I got through the entire install with pretty much no issues, and was able to generate the test image on my cpu. I have a GTX 980 though, and would like to take advantage of it. So I attempted to run the steps to get cuda working, and successfully installed the cuda backend for my gpu and can run `nvidia-smi`. But when I run the command `luarocks install cutorch`, it fails with some build errors, noting that I have an unsupported GNU version, and that gcc versions at or above 4.9 are unsupported. 

As an aside, I'm running Ubuntu 15.10, but I installed the cuda driver for 15.04. It seems to have worked fine regardless, but I don't know if that's related to the issue.

Here is an example snippet of one of the error messages I got alongside the one above;
`CMake Error at THC_generated_THCReduceApplyUtils.cu.o.cmake:206 (message):
  Error generating
  /tmp/luarocks_cutorch-scm-1-1856/cutorch/build/lib/THC/CMakeFiles/THC.dir//./THC_generated_THCReduceApplyUtils.cu.o
`

**Comment by joshicola** [comment]
Maurdekye,
I had success with doing the following on ubuntu before installing cutorch:
export CXX=/usr/bin/g++-4.9
export CC=/usr/bin/gcc-4.9

**Comment by Maurdekye** [comment]
Yeah, I already solved my issue a while ago by doing something similar. I just renamed the gcc versions in my bin so that the default was 4.9 instead of 5.2. A bit more awkward than what you're explaining, but it got the job done. Also, I didn't know that there were environment variables controlling the default gcc at the time.
